lib/open_mic.rb

```diff
@@ -0,0 +1,45 @@
+class OpenMic
+  attr_reader :location, :date, :performers
+
+  def initialize(attrs)
+    @location = attrs[:location]
+    @date = attrs[:date]
+    @performers = []
+  end
+
+  def welcome(user)
+    @performers.push(user)
+  end
+
+  def repeated_jokes?
+    # Gather all known joke ids into array and remove nesting
+    all_joke_ids = joke_ids.values.flatten!
+
+    # If all elements are unique, return false
+    if all_joke_ids.uniq.length == all_joke_ids.length
+      false
+    else
+      true
+    end
+  end
+
+  def joke_ids
```

Comment:
Here, you are trying to get all ids for all jokes for all performers; it would be just as useful (and a bit easier), to get an array of all jokes, regardless of the performer they belong to (since we are flattening out this information above anyway).  A potential simplification of these two methods could look like this:

```ruby
def repeated_jokes?
  all_jokes.uniq.length != all_jokes.length
end

def all_jokes
  @performers.map do |user|
    user.jokes
  end.flatten
end
```